src/jit/gentree.cpp

```diff
@@ -4691,6 +4691,16 @@ bool                GenTree::OperMayThrow()
 
         return true;
 
+    case GT_INTRINSIC:
+        // If this is an intrinsic that represents the object.GetType(), it can throw an NullReferenceException.
+        // Report it as may throw.
+        if (AsIntrinsic()->gtIntrinsicId == CORINFO_INTRINSIC_Object_GetType)
```

Comment:
This code should be more generic by checking for GTF_EXCEPT flag on GT_INTRINSIC node

return (gtFlags & GTF_EXCEPT) != 0;

In future when we have another intrinsic that can also throw an exception, it gets taken care.